src/Loader/StrictPoLoader.php

```diff
@@ -0,0 +1,349 @@
+<?php
+declare(strict_types = 1);
+
+namespace Gettext\Loader;
+
+use Exception;
+use Gettext\Translation;
+use Gettext\Translations;
+
+/**
+ * Class to load a PO file.
+ */
+final class StrictPoLoader extends Loader
+{
+    /** @var Translations */
+    private $translations;
+    /** @var Translation */
+    private $translation;
+    private $data;
+    private $position;
+    private $inPreviousComment;
+
+    public function loadString(string $string, Translations $translations = null): Translations
+    {
+        $this->data = $string;
+        $this->position = 0;
+        $this->translations = parent::loadString($string, $translations);
+        for ($this->newEntry(); $this->getChar() !== null; $this->newEntry()) {
+            while ($this->readComment());
+            $this->readContext();
+            $this->readId();
```

Comment:
FYI here there's one different behavior from the GNU's `msgfmt` related to simple comments (`#`, `#.`, `#,`, `#:`) dangling in the end of the file, `msgfmt` ignores them, while this parser will raise an exception.

Reply:
Looking at this code, it seems like the order of the elements matters. Every translation need the comments first, then the context, id, plurals and translation in this specific order. Is that true?

Reply:
I remembered about my last observation :)
If you use the `#,` with unknown flags (e.g. `#,test`), the comment is dropped... It only keeps the known ones, such as `java-printf-format`

I just didn't investigate the behavior of the `#:`, so I'm not sure if it can be wiped off/generate errors when invalid.

Reply:
Ok. I prefer PoLoader to be more permisive than `msgfmt`. I mean, accept code that would fail in `msgfmt` (or other tools) and output code that is compatible with these tools, like a kind of "fixer".
I think that the order of the elements shouldn't be important.
I don't know if its easy to make this new StrictPoLoader more lax. If not, I wouldn't replace the existing PoLoader (althought both loaders can be included).

Reply:
Yes, I'm agree.
I'm fine including both.
Looking forward the tests before merge it.

Reply:
Ok! I'll write some tests :)

Reply:
Ok, I fixed the broken tests and added some for the other loader.

Reply:
Please, give me one more run 😅

Reply:
The tests passed (both loaders are sharing the tests that you've wrote)...

Tomorrow I'll take a look at the performance, then feel free to merge or ask fixes.

Given the amount of checks that I've added, I guess it will be difficult to make a miracle, but there's room for improvements...

Reply:
Nice, thanks!

Let me know when you are confortable with the code, so I can review and merge :)